tests/test_ble_interface_core.py

```
@@ -4971,6 +4973,252 @@ def test_find_device_direct_connect_without_discovery_manager() -> None:
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    """Notification error reporting should handle hook failures and missing hooks."""
+    iface = object.__new__(BLEInterface)
+    debug_calls: list[str] = []
+    monkeypatch.setattr(
+        "meshtastic.interfaces.ble.interface.logger.debug",
+        lambda message, *_args, **_kwargs: debug_calls.append(cast(str, message)),
+    )
+
+    def _raising_report(_message: str) -> None:
+        raise RuntimeError("hook failed")
```

Ruff (0.15.6)

[warning] Avoid specifying long messages outside the exception class

(TRY003)